File: backend/src/main/java/com/example/backend/service/DiseaseService.java
package com.example.backend.service;

import com.example.backend.model.Disease;
import com.example.backend.model.exceptions.DiseaseDoesNotExist;

import java.util.Optional;

public interface DiseaseService {
    Optional<Disease> getDetailsAboutDisease(String diseaseName) throws DiseaseDoesNotExist;
    String getDiseaseAbstract();
}

File: frontend/src/services/DiseaseService.js
import axios from "../custom-axios/axios"

const DiseaseService={
    getDiseaseAbstract:()=>{
        return axios.get('/disease')
    },
    getDetailsAboutDisease:(name)=>{
        return axios.get(`/disease/${name}`)
    }
}
export default DiseaseService;